src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3301,110 +3219,89 @@ internal virtual ImmutableArray<LabelSymbol> Labels
+                    analyzedArguments,
+                    bindingDiagnostics,
+                    queryClause: null,
+                    allowUnexpandedForm: false);
 
-            return patternMethod;
-        }
+                analyzedArguments.Free();
 
-        /// <summary>
-        /// Finds a pattern method implemented as an extension method, stopping at the first matching method
-        /// </summary>
-        /// <param name="typeExpr">The expression on which to search.</param>
-        /// <param name="methodName">Method to search for.</param>
-        /// <param name="syntaxExpr">The expression for which lookup is being performed</param>
-        /// <param name="diagnostics">Populated with binding diagnostics.</param>
-        /// <param name="messageID">The ID of the message to use when reporting diagnostics</param>
-        /// <returns>The desired method or null.</returns>
-        private MethodSymbol FindPatternExtensionMethod(BoundExpression typeExpr, string methodName, SyntaxNode syntaxExpr,
-                                                        DiagnosticBag diagnostics, MessageID messageID)
-        {
-            AnalyzedArguments arguments = AnalyzedArguments.GetInstance();
-            MethodSymbol patternMethod = null;
+                if (patternMethodCall.Kind != BoundKind.Call)
+                {
+                    // did not find anything callable
+                    return PatternLookupResult.NotCallable;
+                }
 
-            var resolution = BindExtensionMethod(syntaxExpr, methodName, arguments, typeExpr, typeArguments: default, isMethodGroupConversion: false, returnRefKind: RefKind.None, returnType: null);
-            if (resolution.HasAnyApplicableMethod)
-            {
-                if (resolution.OverloadResolutionResult.Succeeded)
+                var call = (BoundCall)patternMethodCall;
+                if (call.ResultKind == LookupResultKind.Empty)
                 {
-                    patternMethod = resolution.OverloadResolutionResult.ValidResult.Member;
+                    // did not find any methods that even remotely fit
+                    return PatternLookupResult.NoResults;
                 }
-                else
+
+                // we have succeeded or almost succeded to bind the method
+                // report additional binding diagnostics that we have seen so far
+                diagnostics.AddRange(bindingDiagnostics);
+
+                var patternMethodSymbol = call.Method;
+                if (patternMethodSymbol is ErrorMethodSymbol ||
+                    patternMethodCall.HasAnyErrors)
                 {
-                    diagnostics.Add(ErrorCode.WRN_PatternIsAmbiguous, syntaxExpr.Location, typeExpr.Type, messageID.Localize(),
-                                           resolution.OverloadResolutionResult.Results[0].Member, resolution.OverloadResolutionResult.Results[1].Member);
+                    // bound to something uncallable
```

Comment:
>uncallable [](start = 42, length = 10)

The comment seems out of date, as it seems to better describe `return PatternLookupResult.NotCallable;`